sway-core/src/declaration_engine/declaration_wrapper.rs

```diff
@@ -69,11 +71,47 @@ impl CopyTypes for DeclarationWrapper {
     }
 }
 
+impl MonomorphizeHelper for DeclarationWrapper {
+    fn type_parameters(&self) -> &[TypeParameter] {
+        match self {
+            DeclarationWrapper::Trait(_)
+            | DeclarationWrapper::TraitFn(_)
+            | DeclarationWrapper::TraitImpl(_)
+            | DeclarationWrapper::Storage(_)
+            | DeclarationWrapper::Unknown => {
+                panic!("declaration type does not support type parameters")
```

Comment:
In general, we should not use `panic!` (`unwrap!`, etc), unless absolutely necessary. The compiler is designed so that if there is an error when type checking one AST nodes, the other AST nodes are relatively unaffected and can continue to be type checked, providing the best end-user experience. `panic!` generates an early exit, preventing the compiler from offering this feature to users. This is why we have the heafty `CompileResult` struct. In this particular case, remove this `panic` and I recommend that you return an empty vec instead.